include/GafferImage/ImageNode.h

```diff
@@ -121,6 +122,7 @@ class GAFFERIMAGE_API ImageNode : public Gaffer::ComputeNode
 		void compute( Gaffer::ValuePlug *output, const Gaffer::Context *context ) const override;
 		/// Compute methods for the individual children of outPlug() - these must be implemented by derived classes, or
 		/// an input connection must be made to the plug, so that the method is not called.
+		virtual IECore::ConstStringVectorDataPtr computeViewNames( const Gaffer::Context *context, const ImagePlug *parent ) const;
```

Comment:
Hmm, good question.  I guess I probably did just imitate channelNames.  The argument that it would make sense to switch both at the same time is reasonable - but on the other side, it would be quicker to switch now when there's no compatibility concerns, and if we did viewNames this way, and it works smoothly, it would be a good demonstration that it would work well to switch channelNames ( which would be a lot more work, because of compatibility and channelNames being used in more places )

Reply:
Summary of separate conversation : we'll get this PR across the line using strings for view name, and then do a quick follow up to use InternedString.